src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -193,14 +191,17 @@ - (void)startNextRead {
   }
   __weak GRPCCall *weakSelf = self;
   __weak GRPCDelegateWrapper *weakWriteable = _responseWriteable;
-
+  
   dispatch_async(_callQueue, ^{
-    [weakSelf startReadWithHandler:^(grpc_event *event) {
-      if (!event->data.read) {
-        // No more responses from the server.
+    [weakSelf startReadWithHandler:^(NSDictionary *event) {
+      NSData *data = event[[NSNumber numberWithInt:GRPC_OP_RECV_MESSAGE]];
+      if (data == [NSNull null]) {
```

Comment:
if (!data || data == [NSNull null]), return
But then, why do we put [NSNull null] there?